mkosi

```diff
@@ -832,6 +833,99 @@ gpgkey={gpg_key}
     with mount_api_vfs(args, workspace):
         subprocess.run(cmdline, check=True)
 
+@complete_step('Installing Mageia')
+def install_mageia(args, workspace, run_build_script):
+
+    disable_kernel_install(args, workspace)
+
+    # Mageia does not (yet) have RPM GPG key on the web
+    gpg_key = '/etc/pki/rpm-gpg/RPM-GPG-KEY-Mageia'
+    if os.path.exists(gpg_key):
+        gpg_key = "file://%s" % gpg_key
+#    else:
+#        gpg_key = "https://getfedora.org/static/%s.txt" % FEDORA_KEYS_MAP[args.release]
+
+    if args.mirror:
+        baseurl = "{args.mirror}/distrib/{args.release}/x86_64/media/core/release/".format(args=args)
+        if not check_if_url_exists("%s/media.repo" % baseurl):
+            baseurl = "{args.mirror}/distrib/cauldron/x86_64/media/core/release/".format(args=args)
+
+        release_url = "baseurl=%s" % baseurl
+        updates_url = "baseurl={args.mirror}/distrib/{args.release}/x86_64/media/core/updates/".format(args=args)
+    else:
+        release_url = ("mirrorlist=https://www.mageia.org/mirrorlist/?" +
+                       "release={args.release}&arch=x86_64&section=core&repo=release".format(args=args))
+        updates_url = ("mirrorlist=https://www.mageia.org/mirrorlist/?" +
+                       "release={args.release}&arch=x86_64&section=core&repo=updates".format(args=args))
+
+    with open(os.path.join(workspace, "dnf.conf"), "w") as f:
+        f.write("""\
+[main]
+gpgcheck=1
+
+[mageia]
+name=Mageia {args.release} Core Release
+{release_url}
+gpgkey={gpg_key}
+
+[updates]
+name=Mageia {args.release} Core Updates
+{updates_url}
+gpgkey={gpg_key}
+""".format(args=args,
+           gpg_key=gpg_key,
+           release_url=release_url,
+           updates_url=updates_url))
+    if args.repositories:
+        repos = ["--enablerepo=" + repo for repo in args.repositories]
+    else:
+        repos = ["--enablerepo=mageia", "--enablerepo=updates"]
+
+    root = os.path.join(workspace, "root")
+    cmdline = ["dnf",
+               "-y",
+               "--config=" + os.path.join(workspace, "dnf.conf"),
+               "--best",
+               "--allowerasing",
+               "--releasever=" + args.release,
+               "--installroot=" + root,
+               "--disablerepo=*",
+               *repos,
+               "--setopt=keepcache=1",
+               "--setopt=install_weak_deps=0"]
+
+    # Turn off docs, but not during the development build, as dnf currently has problems with that
+    if not args.with_docs and not run_build_script:
+        cmdline.append("--setopt=tsflags=nodocs")
+
+    cmdline.extend([
+               "install",
+               "systemd",
+               "mageia-release",
+               "passwd"])
+
+    if args.packages is not None:
+        cmdline.extend(args.packages)
+
+    if run_build_script and args.build_packages is not None:
+        cmdline.extend(args.build_packages)
+
+    if args.bootable:
+        cmdline.extend(["kernel", "binutils"])
```

Comment:
There are several possible `kernel` flavors that could be picked. I'd suggest picking a flavor and sticking with it here. Probably `kernel-server-latest` is the right choice here, as it works the best across a wide variety of targets.